src/main/java/io/stargate/sgv2/jsonapi/service/resolver/CreateIndexCommandResolver.java

```diff
@@ -0,0 +1,128 @@
+    // Validate Column present in Table
+    final Optional<ColumnMetadata> column =
+        ctx.schemaObject()
+            .tableMetadata()
+            .getColumn(CqlIdentifierUtil.cqlIdentifierFromUserInput(columnName));
+    ColumnMetadata columnMetadata =
+        column.orElseThrow(
+            () ->
+                SchemaException.Code.INVALID_INDEX_DEFINITION.get(
+                    Map.of("reason", "Column not defined in the table")));
+    Boolean caseSensitive = definitionOptions != null ? definitionOptions.caseSensitive() : null;
+    Boolean normalize = definitionOptions != null ? definitionOptions.normalize() : null;
+    Boolean ascii = definitionOptions != null ? definitionOptions.ascii() : null;
+    SimilarityFunction similarityFunction =
+        definitionOptions != null ? definitionOptions.metric() : null;
+    String sourceModel = definitionOptions != null ? definitionOptions.sourceModel() : null;
+    if (definitionOptions != null) {
+      // Validate Options
+      if (!columnMetadata.getType().equals(DataTypes.TEXT)) {
+        if (caseSensitive != null || normalize != null || ascii != null) {
+          throw SchemaException.Code.INVALID_INDEX_DEFINITION.get(
+              Map.of(
+                  "reason",
+                  "`caseSensitive`, `normalize` and `ascii` options are valid only for text column"));
+        }
+      }
+      if (!(columnMetadata.getType() instanceof VectorType)) {
+        if (similarityFunction != null || sourceModel != null) {
+          throw SchemaException.Code.INVALID_INDEX_DEFINITION.get(
+              Map.of(
+                  "reason",
+                  "`metric` and `sourceModel` options are valid only for vector type column"));
+        }
+      } else {
+        if (similarityFunction != null && sourceModel != null) {
+          throw SchemaException.Code.INVALID_INDEX_DEFINITION.get(
+              Map.of(
+                  "reason",
+                  "Only one of `metric` or `sourceModel` options should be used for vector type column"));
+        }
+        if (sourceModel != null && !supportedSources.contains(sourceModel)) {
+          throw SchemaException.Code.INVALID_INDEX_DEFINITION.get(
+              Map.of(
+                  "reason",
+                  "Invalid source model. Supported source models are: " + supportedSources));
+        }
+      }
+    }
+
+    // Command level option for ifNotExists
+    boolean ifNotExists = false;
+    final CreateIndexCommand.Options commandOptions = command.options();
+    if (commandOptions != null && commandOptions.ifNotExists() != null) {
+      ifNotExists = commandOptions.ifNotExists();
+    }
+
+    // Default Similarity Function to COSINE
+    if (columnMetadata.getType() instanceof VectorType
+        && similarityFunction == null
+        && sourceModel == null) {
+      similarityFunction = SimilarityFunction.COSINE;
+    }
+
+    var attempt =
```

Comment:
Ok: so since pretty much everything is already available, not sure there's even need for Builder is there? Or if there is, should at least pass more of settings (at least "columnName", "indexName", "ifNotExists") to constructor.